rs-matter/src/utils/zbus_proxies/avahi/entry_group.rs

```diff
@@ -0,0 +1,107 @@
+/*
+ *
+ *    Copyright (c) 2020-2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+//! # D-Bus interface proxy for: `org.freedesktop.Avahi.EntryGroup`
+
+use zbus::proxy;
+
+#[proxy(interface = "org.freedesktop.Avahi.EntryGroup", assume_defaults = true)]
+pub trait EntryGroup {
+    /// AddAddress method
+    fn add_address(
+        &self,
+        interface: i32,
+        protocol: i32,
+        flags: u32,
+        name: &str,
+        address: &str,
+    ) -> zbus::Result<()>;
+
+    /// AddRecord method
+    #[allow(clippy::too_many_arguments)]
+    fn add_record(
```

Comment:
As Rust doesn't seem to have named args (which is a bummer!), does it make sense to have a structure here as a param instead of all these args? as a lot of them are the same type (ints), readability when calling this method is gonna be rough

Reply:
Ah, hadn't realized this is semi-auto generated & maps to some underlying impl; cool, that makes sense then, whole thing would certainly be a lot simpler if rust allowed named args; if we expose these interfaces to be widely-used I think it makes sense to have a wrapper around them that's cleaner, but as a generated proxy having it be like this SGTM

Reply:
Yes, I had the "exposure" case in mind hence I wanted this stuff to be complete (full API) first, and then yes - in future it can be made more user friendly.